### PR TITLE
feat: Config for individual view-generators

### PR DESCRIPTION
**Pull request by rish691**
Allows overriding config for individual view-generator

### CODE REVIEW

File: hypertrace-view-generator/helm/templates/view-generation/single-view-gen-configs.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
```
Comment (kotharironak) [inline hypertrace-view-generator/helm/templates/view-generation/single-view-gen-configs.yaml:1]: we don't need this, right?

File: hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml [hunk 2]
```diff
@@ -59,6 +64,14 @@ spec:
             - name: view-gen-service-config
               mountPath: /app/resources/configs/common/staging/application.conf
               subPath: application.conf
+            - name: view-gen-backend-entity-config
```
Comment (kotharironak) [inline hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml:67]: Here as well, we will need a loop,
```
{- range $viewgen :=$group.generator.serviceConfig.viewgenerators }}
        
 {{- end }}
```

Comment (rish691) [reply hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml:67]: There is a loop on line 70

File: hypertrace-view-generator/helm/templates/view-generation/single-view-gen-configs.yaml (hunk 1 above)
Comment (rish691) [reply hypertrace-view-generator/helm/templates/view-generation/single-view-gen-configs.yaml]: Added a loop here

File: hypertrace-view-generator/helm/templates/view-generation/view-gen-individual-config.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{{ $global := . }}
+{{- range $k, $val := .Values.viewGenIndividualConfigs }}
```
Comment (kotharironak) [inline hypertrace-view-generator/helm/templates/view-generation/view-gen-individual-config.yaml:2]: Is it possible to use same iterator - `$group.generator.serviceConfig.viewgenerators` ?

File: hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml [hunk 4]
```diff
@@ -36,6 +36,11 @@ spec:
         - name: log4j-config
           configMap:
             name: {{ $global.Values.logConfig.name }}
+      {{- range $viewgen := $group.generator.serviceConfig.viewgenerators }}
+        - name: {{ $viewgen }}-config
+          configMap:
+            name: {{ $viewgen }}-config
```
Comment (kotharironak) [inline hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml:42]: It seems to me that names are not matching.
configMap - backend-entity-view-generator-config
viewGenName - view-gen-backend-entity-config

File: hypertrace-view-generator/helm/templates/view-generation/view-gen-individual-config.yaml (hunk 3 above)
Comment (rish691) [reply hypertrace-view-generator/helm/templates/view-generation/view-gen-individual-config.yaml:2]: That iterator is for groups, this iterator is for individual view-gens. Groups are independant from individual config.

File: hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml (hunk 4 above)
Comment (rish691) [reply hypertrace-view-generator/helm/templates/view-generation/view-generator-deployment.yaml:42]: Done

**Review by kotharironak — APPROVED** [review]

**Pull request merged**